src/aux-runtime/runtime/Transpiler.ts

```diff
@@ -301,6 +301,21 @@ export class Transpiler {
         return this._transpile(code);
     }
 
+    /**
+     * Determines if the given node contains any await expressions.
+     * @param node The node.
+     */
+    private _isAsyncNode(node: any): boolean {
+        for (const child of node.body) {
+            if (child.type === 'ExpressionStatement') {
+                if (child.expression.type === 'AwaitExpression') {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
```

Comment:
@MostlyBen do you know if this will cover scenarios like:
```typescript
let result = await os.showInput();
```
or cases where the expression is used inside function arguments:
```typescript
os.download(await web.get('https://example.com'));
```
or other scenarios like using parenthesis, parts of mathematical operations, etc.

Generally, each one of these will parse to different syntax trees. My recommendation would be to traverse the tree, but skip nodes that would cause a new async scope, like function declarations, arrow function declarations, class declarations, etc. Then if you come across any `AwaitExpression` it should mark the script as async. Additionally, this gives us more certainty that we're only changing the behavior in some specific cases.

Finally, I would recommend that you add a couple tests to `Transpiler.spec.ts` so that we can verify that this behavior continues to work in the future (plus it makes it easier for me to review).